tests/7zip_compatiblity_test.rs

```diff
@@ -0,0 +1,115 @@
+#![cfg(feature = "test-7zip")]
+//! This test compresses some random data with deflate64 using p7zip `7z` command and check decompression
+
+use std::ffi::OsString;
+use std::fs::File;
+use deflate64::Deflate64Decoder;
+use std::io::{Cursor, Read, Write};
+use std::mem::size_of;
+use std::process::{Command, Stdio};
+use proptest::proptest;
+use tempfile::TempDir;
+
+const TEST_FILE_NAME: &'static str = "test.file";
+const TEST_ZIP_NAME: &'static str = "test.zip";
+
+#[repr(packed)]
+struct ZipLocalFileHeader {
+    signature: [u8; 4],
+    version: [u8; 2],
+    flags: [u8; 2],
+    compression_method: [u8; 2],
+    last_mod_time: [u8; 2],
+    last_mod_date: [u8; 2],
+    uncompressed_crc32: [u8; 4],
+    compressed_size: [u8; 4],
+    uncompressed_size: [u8; 4],
+    file_name_len: [u8; 2],
+    extra_field_len: [u8; 2],
+}
+
+impl ZipLocalFileHeader {
+    const SIGNATURE: [u8; 4] = [0x50, 0x4B, 0x03, 0x04];
+
+    fn zero() -> Self {
+        unsafe { std::mem::zeroed() }
+    }
+
+    fn as_mut_bytes(&mut self) -> &mut [u8] {
+        unsafe {
+            std::slice::from_raw_parts_mut(
+                self as *mut Self as *mut u8,
+                size_of::<Self>()
+            )
+        }
+    }
+}
+
+fn compress_with_7zip(data: &[u8]) -> Vec<u8> {
+    let mut temp_dir = TempDir::new().unwrap();
+
+    // write data to test file
+    File::create(temp_dir.path().join(TEST_FILE_NAME)).unwrap()
```

Comment:
I think you can have `proptest` generates random number of filenames and then content, that will be more effective at discovering bugs.

Reply:
We're using zip file as container of deflate64 compression so I do not want to make zip reader complicated unnecessary. That's why I'm using constant file name.
Zip file parse is out of scope of deflate64 I think.

Reply:
Aha right, it's unrelated to decompression of deflate64.